pkg/controller/lifecycle/add.go

```diff
@@ -107,7 +107,7 @@ func AddConfigMapWatchingControllerToManager(mgr manager.Manager, options *optio
 	configReconcilerArgs := extension.AddArgs{
 		Actuator:        NewActuator(consts.ActuatorName + consts.ConfigsSuffix),
 		Resync:          60 * time.Minute,
-		FinalizerSuffix: consts.ExtensionType + consts.ConfigsSuffix,
+		FinalizerSuffix: consts.ExtensionType, // on purpuse we're using the same finalizer as original controller to "delete" only once without need for waiting for another "configs" controller
```

Comment:
Sound a bit odd to me. I would propose something like that:
we're using the same finalizer as the original controller on purpose to "delete" only once without a need to wait for another "configs" controller

Reply:
yes that sounds much better.